### PR TITLE
Monitor using a pod monitor

### PR DESCRIPTION
**Pull request by QuentinBisson**
## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

**Comment by QuentinBisson** [comment]
Tested on gollum

File: helm/prometheus-meta-operator/templates/pod-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    name: {{ include "resource.default.name" . }}
+spec:
+  namespaceSelector:
+    matchNames:
+      - {{ include "resource.default.namespace" . }}
+  selector:
+    matchLabels:
+      {{- include "labels.selector" . | nindent 6 }}
+  podMetricsEndpoints:
+    - targetPort: 8000
```
Comment (hervenicol) [inline helm/prometheus-meta-operator/templates/pod-monitor.yaml:16]: Do you have to use the port number there, or could you use its name `http` ?

Comment (QuentinBisson) [reply helm/prometheus-meta-operator/templates/pod-monitor.yaml:16]: We cannot as the port is not exposed in the deployment but that's doable if we expose the port and name it

**Comment by QuentinBisson** [comment]
@giantswarm/team-atlas can I get a quick look?

**Comment by hervenicol** [comment]
The PR could use some more context.

* Why a `podmonitor` and not a `servicemonitor`?
* Wouldn't you want to remove the `service` annotation `giantswarm.io/monitoring`?
* isn't there an issue related to this PR?

**Comment by QuentinBisson** [comment]
> The PR could use some more context.
> 
> * Why a `podmonitor` and not a `servicemonitor`?
Because PMO does not need a Service to run so creating one just for metrics seems useless
> * Wouldn't you want to remove the `service` annotation `giantswarm.io/monitoring`?
Probably yes
> * isn't there an issue related to this PR?
No

**Comment by hervenicol** [comment]
> > * Why a `podmonitor` and not a `servicemonitor`?
>
> Because PMO does not need a Service to run so creating one just for metrics seems useless
>
> > * Wouldn't you want to remove the `service` annotation `giantswarm.io/monitoring`?
>
> Probably yes

Then we can even remove the whole service, if it's useless ?

**Comment by QuentinBisson** [comment]
The service is gone in the PR already :)

**Pull request merged**